File: spring/test/shedlock-springboot-sleuth-test/src/test/java/net/javacrumbs/shedlock/test/boot/IntegrationTest.java
/**
 * Copyright 2009 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.javacrumbs.shedlock.test.boot;

import net.javacrumbs.shedlock.core.LockConfiguration;
import net.javacrumbs.shedlock.core.LockProvider;
import org.junit.jupiter.api.Test;
import org.mockito.ArgumentCaptor;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.mock.mockito.MockBean;

import static org.assertj.core.api.Assertions.assertThat;
import static org.awaitility.Awaitility.await;
import static org.mockito.Mockito.atLeastOnce;
import static org.mockito.Mockito.verify;

@SpringBootTest
class IntegrationTest {

    @MockBean
    private LockProvider lockProvider;

    @Test
    void testScheduler() {
        await().untilAsserted(() -> {
            ArgumentCaptor<LockConfiguration> configCaptor = ArgumentCaptor.forClass(LockConfiguration.class);
            verify(lockProvider, atLeastOnce()).lock(configCaptor.capture());
            assertThat(configCaptor.getValue().getName()).isEqualTo("reportCurrentTime");
        });
    }
}

File: spring/test/shedlock-springboot-kotlin-test/src/test/kotlin/net/javacrumbs/shedlock/test/boot/IntegrationTest.kt
/**
 * Copyright 2009 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package net.javacrumbs.shedlock.test.boot

import com.nhaarman.mockitokotlin2.atLeastOnce
import com.nhaarman.mockitokotlin2.verify
import net.javacrumbs.shedlock.core.LockConfiguration
import net.javacrumbs.shedlock.core.LockProvider
import org.assertj.core.api.Assertions.assertThat
import org.awaitility.Awaitility.await
import org.junit.jupiter.api.Test
import org.mockito.ArgumentCaptor
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.boot.test.mock.mockito.MockBean

@SpringBootTest
class IntegrationTest {
    @MockBean
    private lateinit var lockProvider: LockProvider

    @Test
    fun testScheduler() {
        await().untilAsserted {
            val configCaptor = ArgumentCaptor.forClass(LockConfiguration::class.java)
            verify(lockProvider, atLeastOnce()).lock(configCaptor.capture())
            assertThat(configCaptor.value.name).isEqualTo("reportCurrentTime")
        }
    }
}
